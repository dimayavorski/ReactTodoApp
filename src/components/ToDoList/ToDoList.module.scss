.toDoList {
	margin-top: 40px;
	background-color: white;
	border-radius: 10px;
	padding-top: 1px;
	padding-bottom: 1px;

	.content {
		padding: 5px;
		height: 400px;
		list-style-type: none;

		.draggable {
			cursor: pointer;
		}
	}
}
