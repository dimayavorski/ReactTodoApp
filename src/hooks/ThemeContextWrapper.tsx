import { useEffect, useState } from 'react';
import { ThemeContext, themes } from './ThemeContext';

interface IThemeContextWrapperProps {
	children: JSX.Element;
}
export default function ThemeContextWrapper({
	children,
}: IThemeContextWrapperProps) {
	const [theme, setTheme] = useState<string>(themes.dark);

	function changeTheme(themeState: string) {
		setTheme(themeState);
	}

	useEffect(() => {
		switch (theme) {
			case themes.dark:
				document.body.classList.add('dark-content');
				break;
			case themes.light:
				document.body.classList.remove('dark-content');
				break;
			default:
				document.body.classList.remove('dark-content');
				break;
		}
	}, [theme]);

	return (
		<ThemeContext.Provider value={{ theme, changeTheme }}>
			{children}
		</ThemeContext.Provider>
	);
}
