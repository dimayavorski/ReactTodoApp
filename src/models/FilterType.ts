export enum FilterType {
	ALL = 0,
	COMPLETED = 1,
	INCOMPLETE = 2,
}
