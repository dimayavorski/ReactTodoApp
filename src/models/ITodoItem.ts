export interface ITodoItem {
	id: string;
	text: string;
	checked: boolean;
}
