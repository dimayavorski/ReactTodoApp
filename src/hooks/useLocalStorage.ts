import { ITodoItem } from "../models/ITodoItem";

export function useLocalStorage() {
    const key = "todos";

    const todosMock: ITodoItem[] = [{
        id: new Date().toISOString(),
        text: "12321412",
        checked: false
    },
    {
        id: new Date().toISOString() + "1",
        text: "1asdasdasd",
        checked: false
    }];

    function saveData(todos: ITodoItem[]) {
        localStorage.setItem(key, JSON.stringify(todos));
    }

    function getData(): ITodoItem[] {
        let data = localStorage.getItem(key);
        if (data) {
            let todosInitial: ITodoItem[] = JSON.parse(data);
            return todosInitial;
        }
        return [];
    }

    function initDatabase() {
        console.log("init db.....");
        if (!localStorage.getItem(key))
            localStorage.setItem(key, JSON.stringify(todosMock));
    }

    return {initDatabase, saveData, getData};
}