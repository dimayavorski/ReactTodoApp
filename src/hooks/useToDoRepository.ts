import { ITodoItem } from '../models/ITodoItem';

export function useToDoRepository() {
	const key = 'todos';

	function saveData(todos: ITodoItem[]) {
		localStorage.setItem(key, JSON.stringify(todos));
	}

	function getData(): ITodoItem[] {
		const data = localStorage.getItem(key);

		if (data) {
			const todos: ITodoItem[] = JSON.parse(data) as ITodoItem[];
			return todos;
		}
		return [];
	}

	function getCompletedData(): ITodoItem[] {
		const data = localStorage.getItem(key);

		if (data) {
			const todos = JSON.parse(data) as ITodoItem[];
			return todos.filter((todo) => todo.checked);
		}
		return [];
	}

	function toggleTodo(todo: ITodoItem): ITodoItem[] {
		const data = localStorage.getItem(key);

		if (data) {
			const todos = JSON.parse(data) as ITodoItem[];
			const todoToToggle = todos.find((t) => t.id === todo.id);
			if (todoToToggle) {
				todoToToggle.checked = !todoToToggle.checked;
				saveData(todos);
				return todos;
			}
		}
		return [];
	}

	function getIncompleteData(): ITodoItem[] {
		const data = localStorage.getItem(key);

		if (data) {
			const todos = JSON.parse(data) as ITodoItem[];
			return todos.filter((todo) => todo.checked === false);
		}
		return [];
	}

	function removeCompleted(): ITodoItem[] {
		const data = getIncompleteData();
		saveData(data);
		return data;
	}

	return {
		removeCompleted,
		saveData,
		getData,
		getCompletedData,
		getIncompleteData,
		toggleTodo,
	};
}
