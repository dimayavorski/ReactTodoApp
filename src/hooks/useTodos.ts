import { ITodoItem } from '../models/ITodoItem';
import { useAppDispatch } from './useAppDispatch';
import { removeToDo, updateToDos } from '../store/todoSlice';
import { useToDoRepository } from './useToDoRepository';
import { useAppSelector } from './useAppSelector';
import { FilterType } from '../models/FilterType';

export function useTodos() {
	const {
		saveData,
		getData,
		getCompletedData,
		getIncompleteData,
		toggleTodo,
		removeCompleted,
	} = useToDoRepository();
	const dispatch = useAppDispatch();

	const filter = useAppSelector((state) => state.todos.filter);
	function set(todos: ITodoItem[]) {
		dispatch(updateToDos(todos));
	}
	function getAll() {
		const todos = getData();
		set(todos);
	}

	function filterCompleted() {
		const todos = getCompletedData();
		set(todos);
	}

	function filterIncomplete() {
		const todos = getIncompleteData();
		set(todos);
	}
	function setFiltered() {
		if (filter == FilterType.COMPLETED) {
			filterCompleted();
		} else if (filter == FilterType.INCOMPLETE) {
			filterIncomplete();
		} else {
			getAll();
		}
	}

	function add(text: string, isActive: boolean) {
		const data = getData();
		const todo = {
			id: new Date().toISOString(),
			text,
			checked: isActive,
		};
		data.push(todo);
		saveData(data);
		setFiltered();
	}

	function toggleComplete(todo: ITodoItem) {
		toggleTodo(todo);
		setFiltered();
	}

	function remove(todo: ITodoItem) {
		let data = getData();
		data = data.filter((item) => item.id !== todo.id);
		saveData(data);
		dispatch(removeToDo(todo));
	}

	function clearCompleted() {
		const data = removeCompleted();
		set(data);
	}

	return {
		getAll,
		clearCompleted,
		filterCompleted,
		filterIncomplete,
		set,
		add,
		remove,
		toggleComplete,
		setFiltered,
	};
}
