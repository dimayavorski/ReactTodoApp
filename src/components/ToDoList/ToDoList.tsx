import styles from './ToDoList.module.scss';
import { ToDoItem } from '../ToDoItem/ToDoItem';
import { ITodoItem } from '../../models/ITodoItem';
import { useEffect } from 'react';
import { DragDropContext, Droppable, Draggable } from 'react-beautiful-dnd';
import { useTodos } from '../../hooks/useTodos';
import { useToDoRepository } from '../../hooks/useToDoRepository';
import { useAppSelector } from '../../hooks/useAppSelector';
import { Footer } from '../Footer/Footer';
import { FilterType } from '../../models/FilterType';

export function ToDoList() {
	const todos: ITodoItem[] = useAppSelector((state) => state.todos.todos);
	const filter: number = useAppSelector((state) => state.todos.filter);
	const { set } = useTodos();
	const { getData, saveData } = useToDoRepository();

	const isDraggableDisabled = filter !== FilterType.ALL;
	const onDragEnd = (event: any) => {
		if (!event.destination) {
			return;
		}

		const items = Array.from(todos);
		const [reorderedItem] = items.splice(event.source.index, 1);
		items.splice(event.destination.index, 0, reorderedItem);
		saveData(items);
		set(items);
	};

	useEffect(() => {
		const initData = getData();
		set(initData);
	}, []);

	return (
		<div className={styles.toDoList}>
			<DragDropContext onDragEnd={onDragEnd}>
				<Droppable droppableId="todos">
					{(provided: any) => (
						<ul
							className={styles.content}
							data-testid="content"
							{...provided.droppableProps}
							ref={provided.innerRef}
						>
							{todos.map((todo: ITodoItem, index: number) => (
								<Draggable
									key={todo.id}
									draggableId={todo.id}
									index={index}
									isDragDisabled={isDraggableDisabled}
								>
									{(providedInner: any) => (
										<li
											ref={providedInner.innerRef}
											{...providedInner.draggableProps}
											{...providedInner.dragHandleProps}
										>
											<ToDoItem key={todo.id} todoItem={todo} />
										</li>
									)}
								</Draggable>
							))}
							{provided.placeholder}
						</ul>
					)}
				</Droppable>
			</DragDropContext>
			<Footer />
		</div>
	);
}
