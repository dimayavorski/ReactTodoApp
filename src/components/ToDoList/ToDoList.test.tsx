import { ITodoItem } from '../../models/ITodoItem';
import {
	fireEvent,
	render,
	screen,
	RenderResult,
} from '@testing-library/react';

import { Provider } from 'react-redux';
import { store } from '../../store/store';
import { ToDoList } from './ToDoList';
import { useLocalStorage } from '../../hooks/useToDoRepository';

const mockItems: ITodoItem[] = [
	{
		id: 'asdasd',
		text: 'someName',
		checked: true,
	},
	{
		id: 'asdasd2',
		text: 'someName2',
		checked: true,
	},
	{
		id: 'asdasd3',
		text: 'someName3',
		checked: false,
	},
];
describe('ToDoList tests', () => {
	const { saveData } = useLocalStorage();
	let component: RenderResult;

	saveData(mockItems);
	beforeEach(() => {
		component = render(
			<Provider store={store}>
				<ToDoList />
			</Provider>
		);
	});

	it('should match snapshot', () => {
		expect(component).toMatchSnapshot();
	});

	it('should render component with 3 items', () => {
		expect(screen.queryAllByText(/someName/).length).toBe(3);
	});

	it('should show all todos', () => {
		const allspan = screen.getByText('All');

		fireEvent.click(allspan);

		expect(screen.queryAllByText(/someName/).length).toBe(3);
	});

	it('should show completed todos', () => {
		const allspan = screen.getByText('Completed');

		fireEvent.click(allspan);

		expect(screen.queryAllByText(/someName/).length).toBe(2);
	});

	it('should show incomplete todos', () => {
		const allspan = screen.getByText('Active');

		fireEvent.click(allspan);

		expect(screen.queryAllByText(/someName/).length).toBe(1);
	});
});
