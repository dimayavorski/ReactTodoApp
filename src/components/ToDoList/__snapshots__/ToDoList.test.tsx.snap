// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ToDoList tests should match snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="toDoList"
      >
        <ul
          class="content"
          data-rbd-droppable-context-id="0"
          data-rbd-droppable-id="todos"
          data-testid="content"
        >
          <li
            aria-describedby="rbd-hidden-text-0-hidden-text-0"
            data-rbd-drag-handle-context-id="0"
            data-rbd-drag-handle-draggable-id="asdasd"
            data-rbd-draggable-context-id="0"
            data-rbd-draggable-id="asdasd"
            draggable="false"
            role="button"
            tabindex="0"
          >
            <div
              class="toDoItem"
            >
              <div
                class="toDoItemInfoChecked"
              >
                <div
                  class="checkBoxContainer"
                  data-testid="checkBoxContainer"
                >
                  <label>
                    <input
                      checked=""
                      type="checkbox"
                    />
                    <span />
                  </label>
                </div>
                <p>
                  someName
                </p>
              </div>
              <button
                class="removeBtn"
              >
                <svg
                  fill="currentColor"
                  height="20"
                  stroke="currentColor"
                  stroke-width="0"
                  viewBox="0 0 1024 1024"
                  width="20"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M563.8 512l262.5-312.9c4.4-5.2.7-13.1-6.1-13.1h-79.8c-4.7 0-9.2 2.1-12.3 5.7L511.6 449.8 295.1 191.7c-3-3.6-7.5-5.7-12.3-5.7H203c-6.8 0-10.5 7.9-6.1 13.1L459.4 512 196.9 824.9A7.95 7.95 0 0 0 203 838h79.8c4.7 0 9.2-2.1 12.3-5.7l216.5-258.1 216.5 258.1c3 3.6 7.5 5.7 12.3 5.7h79.8c6.8 0 10.5-7.9 6.1-13.1L563.8 512z"
                  />
                </svg>
              </button>
            </div>
          </li>
          <li
            aria-describedby="rbd-hidden-text-0-hidden-text-0"
            data-rbd-drag-handle-context-id="0"
            data-rbd-drag-handle-draggable-id="asdasd2"
            data-rbd-draggable-context-id="0"
            data-rbd-draggable-id="asdasd2"
            draggable="false"
            role="button"
            tabindex="0"
          >
            <div
              class="toDoItem"
            >
              <div
                class="toDoItemInfoChecked"
              >
                <div
                  class="checkBoxContainer"
                  data-testid="checkBoxContainer"
                >
                  <label>
                    <input
                      checked=""
                      type="checkbox"
                    />
                    <span />
                  </label>
                </div>
                <p>
                  someName2
                </p>
              </div>
              <button
                class="removeBtn"
              >
                <svg
                  fill="currentColor"
                  height="20"
                  stroke="currentColor"
                  stroke-width="0"
                  viewBox="0 0 1024 1024"
                  width="20"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M563.8 512l262.5-312.9c4.4-5.2.7-13.1-6.1-13.1h-79.8c-4.7 0-9.2 2.1-12.3 5.7L511.6 449.8 295.1 191.7c-3-3.6-7.5-5.7-12.3-5.7H203c-6.8 0-10.5 7.9-6.1 13.1L459.4 512 196.9 824.9A7.95 7.95 0 0 0 203 838h79.8c4.7 0 9.2-2.1 12.3-5.7l216.5-258.1 216.5 258.1c3 3.6 7.5 5.7 12.3 5.7h79.8c6.8 0 10.5-7.9 6.1-13.1L563.8 512z"
                  />
                </svg>
              </button>
            </div>
          </li>
          <li
            aria-describedby="rbd-hidden-text-0-hidden-text-0"
            data-rbd-drag-handle-context-id="0"
            data-rbd-drag-handle-draggable-id="asdasd3"
            data-rbd-draggable-context-id="0"
            data-rbd-draggable-id="asdasd3"
            draggable="false"
            role="button"
            tabindex="0"
          >
            <div
              class="toDoItem"
            >
              <div
                class="toDoItemInfo"
              >
                <div
                  class="checkBoxContainer"
                  data-testid="checkBoxContainer"
                >
                  <label>
                    <input
                      type="checkbox"
                    />
                    <span />
                  </label>
                </div>
                <p>
                  someName3
                </p>
              </div>
              <button
                class="removeBtn"
              >
                <svg
                  fill="currentColor"
                  height="20"
                  stroke="currentColor"
                  stroke-width="0"
                  viewBox="0 0 1024 1024"
                  width="20"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M563.8 512l262.5-312.9c4.4-5.2.7-13.1-6.1-13.1h-79.8c-4.7 0-9.2 2.1-12.3 5.7L511.6 449.8 295.1 191.7c-3-3.6-7.5-5.7-12.3-5.7H203c-6.8 0-10.5 7.9-6.1 13.1L459.4 512 196.9 824.9A7.95 7.95 0 0 0 203 838h79.8c4.7 0 9.2-2.1 12.3-5.7l216.5-258.1 216.5 258.1c3 3.6 7.5 5.7 12.3 5.7h79.8c6.8 0 10.5-7.9 6.1-13.1L563.8 512z"
                  />
                </svg>
              </button>
            </div>
          </li>
        </ul>
        <ul
          class="actionsList"
        >
          <li>
            1
             items left
          </li>
          <li
            class="filter"
          >
            <span
              class="footerButton active"
              data-testid="getAll"
            >
              All
            </span>
            <span
              class="footerButton"
              data-testid="getActive"
            >
              Active
            </span>
            <span
              class="footerButton"
              data-testid="getCompleted"
            >
              Completed
            </span>
          </li>
          <li>
            <span
              class="footerButton"
              data-testid="clearCompleted"
            >
              Clear Completed
            </span>
          </li>
        </ul>
      </div>
    </div>
    <div
      aria-atomic="true"
      aria-live="assertive"
      id="rbd-announcement-0"
      style="position: absolute; width: 1px; height: 1px; margin: -1px; border: 0px; padding: 0px; overflow: hidden; clip-path: inset(100%);"
    />
    <div
      id="rbd-hidden-text-0-hidden-text-0"
      style="display: none;"
    >
      
  Press space bar to start a drag.
  When dragging you can use the arrow keys to move the item around and escape to cancel.
  Some screen readers may require you to be in focus mode or to use your pass through key

    </div>
  </body>,
  "container": <div>
    <div
      class="toDoList"
    >
      <ul
        class="content"
        data-rbd-droppable-context-id="0"
        data-rbd-droppable-id="todos"
        data-testid="content"
      >
        <li
          aria-describedby="rbd-hidden-text-0-hidden-text-0"
          data-rbd-drag-handle-context-id="0"
          data-rbd-drag-handle-draggable-id="asdasd"
          data-rbd-draggable-context-id="0"
          data-rbd-draggable-id="asdasd"
          draggable="false"
          role="button"
          tabindex="0"
        >
          <div
            class="toDoItem"
          >
            <div
              class="toDoItemInfoChecked"
            >
              <div
                class="checkBoxContainer"
                data-testid="checkBoxContainer"
              >
                <label>
                  <input
                    checked=""
                    type="checkbox"
                  />
                  <span />
                </label>
              </div>
              <p>
                someName
              </p>
            </div>
            <button
              class="removeBtn"
            >
              <svg
                fill="currentColor"
                height="20"
                stroke="currentColor"
                stroke-width="0"
                viewBox="0 0 1024 1024"
                width="20"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M563.8 512l262.5-312.9c4.4-5.2.7-13.1-6.1-13.1h-79.8c-4.7 0-9.2 2.1-12.3 5.7L511.6 449.8 295.1 191.7c-3-3.6-7.5-5.7-12.3-5.7H203c-6.8 0-10.5 7.9-6.1 13.1L459.4 512 196.9 824.9A7.95 7.95 0 0 0 203 838h79.8c4.7 0 9.2-2.1 12.3-5.7l216.5-258.1 216.5 258.1c3 3.6 7.5 5.7 12.3 5.7h79.8c6.8 0 10.5-7.9 6.1-13.1L563.8 512z"
                />
              </svg>
            </button>
          </div>
        </li>
        <li
          aria-describedby="rbd-hidden-text-0-hidden-text-0"
          data-rbd-drag-handle-context-id="0"
          data-rbd-drag-handle-draggable-id="asdasd2"
          data-rbd-draggable-context-id="0"
          data-rbd-draggable-id="asdasd2"
          draggable="false"
          role="button"
          tabindex="0"
        >
          <div
            class="toDoItem"
          >
            <div
              class="toDoItemInfoChecked"
            >
              <div
                class="checkBoxContainer"
                data-testid="checkBoxContainer"
              >
                <label>
                  <input
                    checked=""
                    type="checkbox"
                  />
                  <span />
                </label>
              </div>
              <p>
                someName2
              </p>
            </div>
            <button
              class="removeBtn"
            >
              <svg
                fill="currentColor"
                height="20"
                stroke="currentColor"
                stroke-width="0"
                viewBox="0 0 1024 1024"
                width="20"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M563.8 512l262.5-312.9c4.4-5.2.7-13.1-6.1-13.1h-79.8c-4.7 0-9.2 2.1-12.3 5.7L511.6 449.8 295.1 191.7c-3-3.6-7.5-5.7-12.3-5.7H203c-6.8 0-10.5 7.9-6.1 13.1L459.4 512 196.9 824.9A7.95 7.95 0 0 0 203 838h79.8c4.7 0 9.2-2.1 12.3-5.7l216.5-258.1 216.5 258.1c3 3.6 7.5 5.7 12.3 5.7h79.8c6.8 0 10.5-7.9 6.1-13.1L563.8 512z"
                />
              </svg>
            </button>
          </div>
        </li>
        <li
          aria-describedby="rbd-hidden-text-0-hidden-text-0"
          data-rbd-drag-handle-context-id="0"
          data-rbd-drag-handle-draggable-id="asdasd3"
          data-rbd-draggable-context-id="0"
          data-rbd-draggable-id="asdasd3"
          draggable="false"
          role="button"
          tabindex="0"
        >
          <div
            class="toDoItem"
          >
            <div
              class="toDoItemInfo"
            >
              <div
                class="checkBoxContainer"
                data-testid="checkBoxContainer"
              >
                <label>
                  <input
                    type="checkbox"
                  />
                  <span />
                </label>
              </div>
              <p>
                someName3
              </p>
            </div>
            <button
              class="removeBtn"
            >
              <svg
                fill="currentColor"
                height="20"
                stroke="currentColor"
                stroke-width="0"
                viewBox="0 0 1024 1024"
                width="20"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M563.8 512l262.5-312.9c4.4-5.2.7-13.1-6.1-13.1h-79.8c-4.7 0-9.2 2.1-12.3 5.7L511.6 449.8 295.1 191.7c-3-3.6-7.5-5.7-12.3-5.7H203c-6.8 0-10.5 7.9-6.1 13.1L459.4 512 196.9 824.9A7.95 7.95 0 0 0 203 838h79.8c4.7 0 9.2-2.1 12.3-5.7l216.5-258.1 216.5 258.1c3 3.6 7.5 5.7 12.3 5.7h79.8c6.8 0 10.5-7.9 6.1-13.1L563.8 512z"
                />
              </svg>
            </button>
          </div>
        </li>
      </ul>
      <ul
        class="actionsList"
      >
        <li>
          1
           items left
        </li>
        <li
          class="filter"
        >
          <span
            class="footerButton active"
            data-testid="getAll"
          >
            All
          </span>
          <span
            class="footerButton"
            data-testid="getActive"
          >
            Active
          </span>
          <span
            class="footerButton"
            data-testid="getCompleted"
          >
            Completed
          </span>
        </li>
        <li>
          <span
            class="footerButton"
            data-testid="clearCompleted"
          >
            Clear Completed
          </span>
        </li>
      </ul>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
